File: getHostname.scala
object HelloWorld {
    
    def processReferer(url : String): String = {
        var referer : String = url
        if (referer.startsWith("http:")){
          referer = referer.substring(5).replaceAll("^[/]+|[/]+$", "")
          return processReferer(referer)
        }
        if (referer.startsWith("https:")){
          referer = referer.substring(6).replaceAll("^[/]+|[/]+$", "")
          return processReferer(referer)
        }
        if (referer.startsWith("ftp:")){
          referer = referer.substring(4).replaceAll("^[/]+|[/]+$", "")
          return processReferer(referer)
        }
        if (referer.startsWith(":")){
          referer = referer.substring(1).replaceAll("^[/]+|[/]+$", "")
          return processReferer(referer)
        }
        referer = referer.replaceAll("//","/")
        if(referer.startsWith("/")) referer.split("/")(1)
        else referer.split("/")(0)
    }
      
    
   def main(args: Array[String]) {
      
      
        var str : String = "http://yvp.yumenetworks.com/[[IMPORT]]/njpl.yumenetworks.com/include/advertiser/http://component_as3.swf16" 
        var str1 : String = "http:www.taptica.com60"
        var str2 : String = "http:/53"
        var str3 : String = "https://https://secure.netflix.com/us/tvui/release-2016_02_29-46/13_3/sapphire/720p/gibbon/signupwizard.js?nrdjsSuffix=nrdjs%2Fv2.21.7-23"
        var str4 : String = "://yytingting.com62"
        var str5 : String = "https://www-m.cnn.com/2018/05/23/europe/yulia,skripal-nerve-agent-recovery-intl/index.html49"
        var str6 : String = "ftp:/abcdefgh.edu/hello"
        var str7 : String = "ftp://http://http://[fdf8:f53b:82e4::53]"
        //String[] arrOfStr = str.split("http://|https://|,",2);
        var hostname : String = processReferer(str7)
        println("..."+hostname+"...")
      
      
      
   }
}

File: getHostname.java
public class getHostname { 
    
    public static String processURLforHostname(String url){
            if (url.startsWith("http:")){
                    url = url.substring(5).replaceAll("^[/]+|[/]+$", "");
                    return processURLforHostname(url);
            }
            if (url.startsWith("https:")){
                    url = url.substring(6).replaceAll("^[/]+|[/]+$", "");
                    return processURLforHostname(url);
            }
            if (url.startsWith("ftp:")){
                    url = url.substring(4).replaceAll("^[/]+|[/]+$", "");
                    return processURLforHostname(url);
            }
            if (url.startsWith(":")){
                    url = url.substring(1).replaceAll("^[/]+|[/]+$", "");
                    return processURLforHostname(url);
            }
            url = url.replaceAll("//","/");
            if(url.startsWith("/")){
                    return processURLforHostname(url.replaceAll("^[/]+|[/]+$", ""));
            }else{
                    return url.split("/")[0];
            }
    }

    
    public static void main(String args[]) 
    { 
        
        
        String str = "http://yvp.yumenetworks.com/[[IMPORT]]/njpl.yumenetworks.com/include/advertiser/http://component_as3.swf16"; 
        String str1 = "www.taptica.com60";
        String str2 = "http:/53";
        String str3 = "https://https://secure.netflix.com/us/tvui/release-2016_02_29-46/13_3/sapphire/720p/gibbon/signupwizard.js?nrdjsSuffix=nrdjs%2Fv2.21.7-23";
        String str4 = "://yytingting.com62";
        String str5 = "/";
        //String[] arrOfStr = str.split("http://|https://|,",2);
        String hostname = processURLforHostname(str4);
        System.out.println(hostname);
  
        // for (String a : arrOfStr){ 
        //     System.out.println(a);
        //     System.out.println("->");
        // }
    } 
}
